tests/db/test_bigchain_api.py

```diff
@@ -56,6 +56,19 @@ def test_create_transaction_payload(self, b, user_vk):
         assert len(tx['transaction']['data']['uuid']) == 36
         assert tx['transaction']['data']['payload'] == payload
 
+    def test_get_transactions_for_payload(self, b, user_vk):
+        payload = {'msg': 'Hello BigchainDB!'}
+        tx = b.create_transaction(b.me, user_vk, None, 'CREATE', payload=payload)
+        payload_uuid = tx['transaction']['data']['uuid']
+
+        matches = b.get_tx_by_payload_uuid(payload_uuid)
```

Comment:
This method will check the `bigchain` table for transactions with that matching payload. At this point you created the transactions but the transaction was not written to the the database. To do so you need to create and write a block.

```python
block = b.create_block([tx])
b.write_block(block, durability='hard')
```